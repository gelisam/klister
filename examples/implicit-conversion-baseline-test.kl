#lang "implicit-conversion-baseline.kl"

(example
  42)

(example
  (let-implicit string-length
    (+ 1 "foo")))

--(example
--  (let-implicit string-length
--    (let-implicit integer->string
--      (string-append (+ 1 "foo") "!"))))
--
--(example
--  (lambda ((the Integer x))
--    (+ 2 x)))
--
--(example
--  (let-implicit (lambda ((the String s))
--                  (* 10 (string-length s)))
--    (let-implicit integer->string
--      (string-append (+ 1 "foo") "!"))))
