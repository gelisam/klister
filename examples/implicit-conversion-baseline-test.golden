42 : Integer
4 : Integer
"4!" : String
#<closure> : (Integer â†’ Integer)
"31!" : String
